### PR TITLE
[FIX] leaves: Accrual leaves on multiple employees fail to update

### PR DESCRIPTION
**Pull request by Julien00859**
When creating an accrual leave for multiple employees (either by
company, by departement or by employee tag), the cron updating
accrual fail because it has been written to just update the leave
durating according to the work time of one employee.

This PR corrects the bug by appliying the accrual leave on every
employee.

opw-1895302

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by Julien00859** [comment]
@nim-odoo @RomainLibert

**Comment by RomainLibert** [comment]
It still lack the support for 'by tag', 'by department' and 'by company' allocations, for which you have to propagate the accrual flag to the children allocations

**Comment by Julien00859** [comment]
Yes, that's the part of your message in the ticket I didn't totally understand

**Comment by Julien00859** [comment]
I didn't use the `_prepare_holiday_values` function because it has the `self.ensure_one()`

File: addons/hr_holidays/models/hr_leave_allocation.py [hunk 1]
```diff
@@ -143,56 +143,74 @@ def _update_accrual(self):
         """
         today = fields.Date.from_string(fields.Date.today())
 
+
```
Comment (nim-odoo) [inline addons/hr_holidays/models/hr_leave_allocation.py]: Remove new white line

File: addons/hr_holidays/models/hr_leave_allocation.py [hunk 2]
```diff
@@ -143,56 +143,74 @@ def _update_accrual(self):
...
-            period_start = datetime.combine(today, time(0, 0, 0)) - delta
-            period_end = datetime.combine(today, time(0, 0, 0))
-
-            # We have to check when the employee has been created
-            # in order to not allocate him/her too much leaves
-            creation_date = fields.Datetime.from_string(holiday.employee_id.create_date)
+            employee_ids = {"company": holiday.mode_company_id.user_ids,
+                            "department": holiday.department_id.member_ids,
+                            "category": holiday.category_id.employee_ids,
+                            "employee": [holiday.employee_id]
```
Comment (nim-odoo) [inline addons/hr_holidays/models/hr_leave_allocation.py]: No need to create a list. We can loop on a recordset with one element, no problem.

**Pull request closed without merging**